### PR TITLE
Test Cloud: support for relative paths in --include argument

### PR DESCRIPTION
**Pull request by kmiecikt**
In the current version, values of the `--include` argument had to use format `targetPath=sourcePath`.
This change allows to use only single path, as long as it is inside "root directory", i.e.:
- `--build-dir` for Xamarin UI Tests, Espresso and Appium
- `--project-dir` for Calabash

This change also improves validation and unit tests for included files parser.

### CODE REVIEW

**Comment by christav** [comment]
@kmiecikt I've merged your other open PRs because they were fairly simple. This one is a little more complex. I took a  look and I think it's fine, but I'd like to have somebody on your team approve as well before merging just because I may have missed something that someone more familiar with testcloud would catch.

File: test/commands/test/lib/included-files-parser-test.ts [hunk 1]
```diff
@@ -1,22 +1,87 @@
...
+  function normalizeFileDescriptions(descriptions: IFileDescriptionJson[]): IFileDescriptionJson[] {
+    return descriptions.map(d => {
+      return {
+        "targetPath": normalizePath(d.targetPath),
+        "sourcePath": normalizePath(d.sourcePath)
+      };
+    })
+  }
+
+  it("should parse pairs with relative target path and valid source path", () => {
```
Comment (skovsboll) [inline test/commands/test/lib/included-files-parser-test.ts]: This one fails on Mac:

```
  1 failing

  1) parseIncludedFiles should parse pairs with relative target path and valid source path:

      AssertionError: expected [ Array(2) ] to deeply equal [ Array(2) ]
      + expected - actual

       [
         {
      -    "sourcePath": "d:/workspace/d:/Temp/Data"
      +    "sourcePath": "d:/Temp/Data"
           "targetPath": "data/foo"
         }
         {
           "sourcePath": "d:/workspace/bar"

      at assertEqual (node_modules/chai/lib/chai/core/assertions.js:485:19)
      at ctx.(anonymous function) (node_modules/chai/lib/chai/utils/addMethod.js:41:25)
      at doAsserterAsyncAndAddThen (node_modules/chai-as-promised/lib/chai-as-promised.js:293:29)
      at .<anonymous> (node_modules/chai-as-promised/lib/chai-as-promised.js:252:17)
      at ctx.(anonymous function) [as equal] (node_modules/chai/lib/chai/utils/overwriteMethod.js:49:33)
      at Context.it (test/commands/test/lib/included-files-parser-test.ts:37:41)
```

File: test/commands/test/lib/included-files-parser-test.ts [hunk 2]
```diff
@@ -1,22 +1,87 @@
...
+      parseIncludedFiles(["/tmp/source"], unitRootDir)
+    }).to.throw();
+  });
+
+  it("should reject string that is not a path", () => {
+    expect(() => {
+      parseIncludedFiles(["|a"], windowsRootDir)
+    }).to.throw();
+  });
 });
```
Comment (skovsboll) [inline test/commands/test/lib/included-files-parser-test.ts]: Here's another edge case you could consider adding:

```
  it("should accept single absolute, deep path under root dir", () => {
    let rawIncludedFiles = [ "/home/user/workspace/somewhere/myAbsoluteData" ];
    let parsedIncludedFiles = normalizeFileDescriptions(parseIncludedFiles(rawIncludedFiles, unitRootDir));

    let expected = [
      {
        "targetPath": "somewhere/myAbsoluteData",
        "sourcePath": "/home/user/workspace/somewhere/myAbsoluteData"
      }
    ];

    expect(parsedIncludedFiles).to.deep.equal(expected);
  });
```

File: src/commands/test/lib/included-files-parser.ts [hunk 3]
```diff
@@ -1,17 +1,68 @@
...
+    return parseIncludedFileFromPathsPair(includedFile, rootDir, separatorIndex);
+  }
+}
+
+function parseIncludedFileFromSinglePath(includedFile: string, rootDir: string): IFileDescriptionJson {
+  validatePath(includedFile);
+
+  if (path.isAbsolute(includedFile)) {
+    let targetPath = path.relative(rootDir, includedFile);
+    if (targetPath.indexOf("..") != -1) {
```
Comment (skovsboll) [inline src/commands/test/lib/included-files-parser.ts]: I don't think you have a test for this case -- trying to escape using `..` ?

File: src/commands/test/lib/included-files-parser.ts [hunk 4]
```diff
@@ -1,17 +1,68 @@
...
+  else {
+    return {
+      targetPath: includedFile,
+      sourcePath: path.join(rootDir, includedFile)
+    }
+  }
+}
+
+function parseIncludedFileFromPathsPair(includedFile: string, rootDir: string, separatorIndex: number): IFileDescriptionJson {
+  let targetPath = includedFile.substr(0, separatorIndex);
```
Comment (skovsboll) [inline src/commands/test/lib/included-files-parser.ts]: Most of the time, using a package for abstracting these path manipulations can be a good idea. For NodeJS, there's https://nodejs.org/docs/latest/api/path.html. Just a suggestion.

Comment (kmiecikt) [inline src/commands/test/lib/included-files-parser.ts]: Yes, I'm already using it. The substr is for parsing "target=source" into two separate paths. Once I have them, I'm using the path module.

File: test/commands/test/lib/included-files-parser-test.ts (hunk 1 above)
Comment (kmiecikt) [inline test/commands/test/lib/included-files-parser-test.ts]: Fixed

File: src/commands/test/lib/included-files-parser.ts (hunk 3 above)
Comment (kmiecikt) [inline src/commands/test/lib/included-files-parser.ts]: Thanks - added this test
